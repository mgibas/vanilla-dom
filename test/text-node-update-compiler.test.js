const compiler = require('../src/text-node-update-compiler')

describe('text-node-update-compiler', () => {
  let options = {state:'st'}

  describe('on compile', () => {
    it('single path', () => {
      expect(compiler.compile('testElement', {
        paths: ['bar'],
        template: () => '"test-template-string"'
      }, options)).toMatchSnapshot();
    })
    it('multiple paths', () => {
      expect(compiler.compile('testElement', {
        paths: ['bar', 'foo'],
        template: () => '"test-template-string"'
      }, options)).toMatchSnapshot();
    })
  })
})
