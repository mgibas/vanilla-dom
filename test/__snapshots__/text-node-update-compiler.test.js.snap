// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text-node-update-compiler on compile multiple paths 1`] = `
"_updaters.push({
  update: st => {
    testElement.nodeValue = 'test-template-string';
  }
});"
`;

exports[`text-node-update-compiler on compile single path 1`] = `
"_updaters.push({
  update: st => {
    testElement.nodeValue = 'test-template-string';
  }
});"
`;
